cpp/include/cudf/io/json.hpp

```diff
@@ -1336,6 +1336,14 @@ class json_writer_options_builder {
 void write_json(json_writer_options const& options,
                 rmm::cuda_stream_view stream = cudf::get_default_stream());
 
+/**
+ * @brief Checks if a cudf::data_type is supported for JSON writing.
+ *
+ * @param type The data_type to check.
+ * @return true if the type is supported for JSON writing, false otherwise.
+ */
+bool is_json_writable_type(data_type type);
```

Comment:
should the new APIs be `constexpr`?

Reply:
See https://github.com/rapidsai/cudf/pull/18468#discussion_r2080463730. Please correct me if I'm wrong

Reply:
I think it could be `constexpr`, since `type_dispatcher` is.

Reply:
Done https://github.com/rapidsai/cudf/pull/18468/commits/1853a3c65d1e9c0bd9bfcfc98b4d096b352cadfd